components/lib/crash/src/main/java/mozilla/components/lib/crash/Breadcrumb.kt

```diff
@@ -97,4 +100,35 @@ data class Breadcrumb(
          */
         USER
     }
+
+    override fun compareTo(other: Breadcrumb): Int {
+        if (this.level.ordinal == other.level.ordinal) {
+            return this.date.compareTo(other.date)
+        }
+
+        return this.level.ordinal.compareTo(other.level.ordinal)
+    }
+}
+
+internal class BreadCrumbPriorityQueue(
+    private val maxSize: Int
+) : PriorityQueue<Breadcrumb>() {
+    override fun add(element: Breadcrumb?): Boolean {
+        val result = super.add(element)
+        if (this.size > maxSize) {
+            this.poll()
+        }
+        return result
+    }
+
+    fun toSortedArrayList(): ArrayList<Breadcrumb> {
+        val breadcrumbsArrayList: ArrayList<Breadcrumb> = arrayListOf()
+        if (isNotEmpty()) {
+            breadcrumbsArrayList.addAll(this)
+
+            /* Sort by timestamp before reporting */
+            breadcrumbsArrayList.sortWith(Comparator { bc0, bc1 -> bc0.date.compareTo(bc1.date) })
```

Comment:
nit: Another option would be to use `sortBy {}` and "map" to a value for sorting (e.g. the date object or the timestamp).

Reply:
I'll look into if sortBy is a better solution.  Thanks,

Reply:
Yes, you're right.  sortBy is so much cleaner.  Tested again and functionality did not change.  Thanks,